test/integration/builder/transaction.js

```diff
@@ -659,5 +659,18 @@ module.exports = function(knex) {
         return builder;
       });
     });
+
+    it('#3690 does not swallow exception when error during transaction occurs', async () => {
+      if (knex.client.driverName !== 'mysql') return;
```

Comment:
@maximelkin : I think we need to remove this conditional from the beginning of the test.  With the conditional removed:

1. The test cases still passes in this branch, and
2. The test case fails in `master`.  (Specifically: against the `mysql` and `mysql2` Clients, which is the behavior we were expecting to see)

Reply:
This possibly not works for non mysql, because each client needs special kill connection command

Reply:
Currently just as proof of concept

Reply:
Ah -- that makes sense, then.  So yeah: we should figure out a good way to run this test case against all of the various Clients.